src/sessions.ts

```diff
@@ -494,16 +499,25 @@ export class ClientSession
 
     try {
       await executeOperation(this.client, operation);
+      this.commitAttempted = undefined;
       return;
     } catch (firstCommitError) {
+      this.commitAttempted = true;
       if (firstCommitError instanceof MongoError && isRetryableWriteError(firstCommitError)) {
         // SPEC-1185: apply majority write concern when retrying commitTransaction
         WriteConcern.apply(command, { wtimeoutMS: 10000, ...wc, w: 'majority' });
         // per txns spec, must unpin session in this case
         this.unpin({ force: true });
 
         try {
-          await executeOperation(this.client, operation);
+          await executeOperation(
+            this.client,
+            new RunAdminCommandOperation(command, {
```

Comment:
Is there a reason for instantiating a new operation here? I see before we just reused the old one.

Reply:
Ah yes, that's what I missed. Thanks!